repository_service_tuf_worker/models/targets/crud.py

```diff
@@ -0,0 +1,149 @@
+# SPDX-FileCopyrightText: 2022 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+from datetime import datetime
+from typing import Any, Dict, List, Tuple
+
+from sqlalchemy.orm import Session
+
+from repository_service_tuf_worker.models.targets import models, schemas
+
+
+def create(db: Session, target: schemas.TargetsCreate):
+    """
+    Create a new Target entry in the DB.
+    """
+    db_target = models.RSTUFTargets(**target.dict())
+    db.add(db_target)
+    db.commit()
+    db.refresh(db_target)
+
+    return db_target
+
+
+def read_unpublished_rolenames(db: Session) -> Tuple[bool, str]:
+    """
+    Read delegated role names that contains unpublished Targets.
+    """
+    return (
+        db.query(models.RSTUFTargets.published, models.RSTUFTargets.rolename)
+        .filter(
+            models.RSTUFTargets.published == False,  # noqa
+        )
+        .distinct()
+        .all()
+    )
+
+
+def read_by_path(db: Session, path: str) -> models.RSTUFTargets:
+    """
+    Read the Target based in the path (unique value).
+    """
+    return (
+        db.query(models.RSTUFTargets)
+        .filter(models.RSTUFTargets.path == path)
+        .first()
+    )
+
+
+def read_by_rolename(db: Session, rolename: str) -> List[models.RSTUFTargets]:
+    """
+    Read all Targets by the delegated role name.
+    """
+    return (
+        db.query(models.RSTUFTargets)
+        .filter(
+            models.RSTUFTargets.rolename == rolename,
+        )
+        .all()
+    )
+
+
+def read_unpublished_by_rolename(
+    db: Session, rolename: str
+) -> List[models.RSTUFTargets]:
+    """
+    Read all unpublished Targets by a delegated role name.
+    """
+    return (
+        db.query(
+            models.RSTUFTargets,
+        )
+        .filter(
+            models.RSTUFTargets.published == False,  # noqa
+            models.RSTUFTargets.rolename == rolename,
+        )
+        .all()
+    )
+
+
+def read_all_add_by_rolename(
+    db: Session, rolename: str
+) -> List[Tuple[str, Dict[str, Any]]]:
+    """
+    Read all delegated role names `path` and `info` that contains targets
+    with action 'ADD'.
+    """
+    return (
+        db.query(
+            models.RSTUFTargets.path,
+            models.RSTUFTargets.info,
+        )
+        .filter(
+            models.RSTUFTargets.rolename == rolename,
+            models.RSTUFTargets.action == schemas.TargetAction.ADD,
+        )
+        .all()
+    )
+
+
+def update(
```

Comment:
This will overwrite the old targets or add new RSTRUFTargets with an "ADD" action and keep the old one?

Reply:
It overwrites the existent RSTFUTargets with ADD. 
Even an existent RSTUFTarget that has action as `REMOVE`, if updated, must move to `ADD`.
A delete/remove action never operates in an existent target with the action 'REMOVE'. It is 'REMOVED' already.

Reply:
> A delete/remove action never operates in an existent target with the action 'REMOVE'. It is 'REMOVED' already.

Sorry, but I am not sure I understood what you mean here.

Reply:
This CRUD is an Update and always overwrites an existent RSTUFTarget with "ADD".
The only case we operate on an existent RSTUFTarget is in `add_targets`

https://github.com/vmware/repository-service-tuf-worker/blob/e5e398ecd601f047b65175f542b09445d08fc905/repository_service_tuf_worker/repository.py#L569-L575

If an RSTUFTarget has an action `REMOVED` and `published` is False, we just reinforce it:

https://github.com/vmware/repository-service-tuf-worker/blob/38ea42cd64ea8b683b00e60da7f37b2f9d3dc806/repository_service_tuf_worker/repository.py#L574-L577